bson/src/main/scala/types.scala

```diff
@@ -1006,9 +1009,9 @@ case class BSONDocument(stream: Stream[Try[BSONElement]])
    * If the value could not be deserialized or converted, returns a `Failure`.
    */
   def getAsUnflattenedTry[T](key: String)(implicit reader: BSONReader[_ <: BSONValue, T]): Try[Option[T]] = getAsTry(key)(reader) match {
-    case Failure(_: DocumentKeyNotFound) => Success(None)
-    case Failure(e)                      => Failure(e)
-    case Success(e)                      => Success(Some(e))
+    case Failure(_: DocumentKeyNotFound) | Failure(_: DocumentKeyIsNull) => Success(None)
```

Comment:
Is it really important to have a distinct exception type? Why not `DocumentKeyNotFound`? Would have this change.

Reply:
No, it is not essential. It did this to have a distinction between the 2 cases, but the end result is the same. I will change that